### PR TITLE
Add comments feature

### PR DESCRIPTION
**Pull request by szymonjanota**

### CODE REVIEW

File: src/components/Button/Button.jsx [hunk 1]
```diff
@@ -65,4 +66,4 @@ Button.defaultProps = {
   children: null,
 };
 
-export default Button;
+export default styled(Button)``;
```
Comment (lukix) [inline src/components/Button/Button.jsx]: Seems like this change is unnecessary

File: src/components/Comments/Comments.jsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+              meta={{}}
+              onChange={this.handleInputChange}
+              disabled={sending}
+            />
+            <Button
+              loading={sending}
+              onClick={this.handleCreate}
+              primary
+            >
+              Wyślij
```
Comment (lukix) [inline src/components/Comments/Comments.jsx]: Use translations

File: src/components/Comments/Comments.jsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+            }
+          </StyledCommentsSection>
+          )
+          }
+        </Card.Body>
+        <Card.Footer>
+          <InputGroup>
+            <Input
+              type="text"
+              placeholder="Skomentuj"
```
Comment (lukix) [inline src/components/Comments/Comments.jsx]: Use translations

File: src/components/Input/Input.jsx [hunk 4]
```diff
@@ -52,4 +52,4 @@ Input.defaultProps = {
   inputRef: () => {},
 };
 
-export default Input;
+export default styled(Input)``;
```
Comment (lukix) [inline src/components/Input/Input.jsx]: Why?

File: src/utils/avatar.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+/* eslint-disable */
```
Comment (lukix) [inline src/utils/avatar.js]: I guess you copied this code (whole file) from somewhere else. Wouldn't it be better to just use some npm package? For example https://www.npmjs.com/package/color-hash

File: src/components/Button/Button.jsx (hunk 1 above)
Comment (marcinlesek) [reply src/components/Button/Button.jsx]: +1

File: src/components/Comments/CommentEntry.jsx [hunk 6]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Text,
+} from 'components';
+import { StyledAvatar, StyledCommentEntry, StyledCommentEntryContent } from './CommentEntry.style';
```
Comment (marcinlesek) [inline src/components/Comments/CommentEntry.jsx]: Maybe block form will be more readable (one under one)?

File: src/components/Comments/CommentEntry.style.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+      margin-left: 20%
+    }
+    border-top-right-radius: 8px
+  ` : `
+    @media (min-width: 480px) { 
+      margin-right: 20%
+    }
+    border-top-left-radius: 8px
+  `)}
+
```
Comment (marcinlesek) [inline src/components/Comments/CommentEntry.style.js]: unnecessary empty line?

File: src/components/Comments/CommentEntry.style.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+import { colors } from 'config';
+import { getAvatarColor } from '../../utils/avatar';
+
+export const StyledCommentEntryContent = styled.div`
+  text-align: left
+  display: inline-block
+  padding: 8px 16px
+  background: ${({ isSelf }) => (isSelf ? colors.primary[0] : colors.gray[2])}
+  border-radius: 20px
+  ${({ isSelf }) => (isSelf ? `
```
Comment (marcinlesek) [inline src/components/Comments/CommentEntry.style.js]: add one empty line to had it more readable?

File: src/components/Comments/CommentEntry.style.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+      margin-right: 20%
+    }
+    border-top-left-radius: 8px
+  `)}
+    
+`;
+
+export const StyledCommentEntry = styled.div`
+  margin-bottom: 10px
+  ${({ isSelf }) => isSelf && `
```
Comment (marcinlesek) [inline src/components/Comments/CommentEntry.style.js]: as above, one empty line before props condition will be more readable.

File: src/components/Comments/Comments.jsx [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+import CommentEntry from './CommentEntry';
+
+class Comments extends PureComponent {
+  state = { value: '' }
+
+  componentDidUpdate(prevProps) {
+    const { sending } = this.props;
+
+    if (prevProps.sending && !sending) {
+      // TODO
```
Comment (marcinlesek) [inline src/components/Comments/Comments.jsx]: hellothere

File: src/components/Comments/Comments.jsx [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+
+  render() {
+    const { value } = this.state;
+    const { comments, sending } = this.props;
+
+    return (
+      <Card fillHeight>
+        <Card.Header>
+          <Text type="subtitle1">
+            Komentarze
```
Comment (marcinlesek) [inline src/components/Comments/Comments.jsx]: Translations

File: src/components/Input/Input.jsx (hunk 4 above)
Comment (marcinlesek) [reply src/components/Input/Input.jsx]: +1

File: src/components/SocialWidget/SocialWidget.jsx [hunk 12]
```diff
@@ -75,6 +75,7 @@ class SocialWidget extends PureComponent {
 }
 
 SocialWidget.propTypes = {
+  lotteryStatic: PropTypes.shape({}).isRequired,
```
Comment (marcinlesek) [inline src/components/SocialWidget/SocialWidget.jsx]: 👍

File: src/components/Button/Button.jsx (hunk 1 above)
Comment (szymonjanota) [reply src/components/Button/Button.jsx]: It is very necessary, cause you [cannot target React components](https://www.styled-components.com/docs/advanced#caveat).

File: src/utils/avatar.js (hunk 5 above)
Comment (szymonjanota) [reply src/utils/avatar.js]: Another dependency? Meh 😀

Comment (szymonjanota) [reply src/utils/avatar.js]: And not whole file - only two first functions 🤣

Comment (lukix) [reply src/utils/avatar.js]: Yep, another dependency would be better.
You can leave it, but then remove eslint-disable comment, fix eslint errors and add tests :)

File: src/components/Comments/Comments.jsx [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+      userInputValue: props.user || '',
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    const { sending } = this.props;
+
+    if (prevProps.sending && !sending) {
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState(() => ({
```
Comment (whereiswolf) [inline src/components/Comments/Comments.jsx]: wat
po co eslint?
a `setState` raczej w ten sposób:
```javascript
this.setState({ value: '' });
```

File: src/components/Comments/Comments.jsx [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+        value: '',
+      }));
+    }
+  }
+
+  handleInputChange = inputName => (e) => {
+    const { value } = e.target;
+
+    this.setState(() => ({
```
Comment (whereiswolf) [inline src/components/Comments/Comments.jsx]: Tak jak w poprzednim komentarzu
```javascript
this.setState({ [inputName]: value });
```

File: src/components/Comments/CommentsProvider.jsx [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+
+const httpClient = axios.create({
+  withCredentials: true,
+});
+
+const { SOCKET_URL, API_URL } = config;
+const socket = openSocket(`${SOCKET_URL}/comments`);
+
+class CommentsProvider extends PureComponent {
+  constructor(props) {
```
Comment (whereiswolf) [inline src/components/Comments/CommentsProvider.jsx]: Możesz używać notacji strzałkowej zamiast konstruktora.

```javascript
state = { user: '', ... }
```

File: src/components/Comments/Comments.jsx [hunk 16]
```diff
@@ -0,0 +1,175 @@
...
+  InputGroup,
+} from 'components';
+import {
+  StyledCommentsSection,
+} from './Comments.style';
+import CommentEntry from './CommentEntry';
+import CurrentUser from './CurrentUser';
+
+class Comments extends PureComponent {
+  constructor(props) {
```
Comment (whereiswolf) [inline src/components/Comments/Comments.jsx]: O tutaj też

**Pull request merged**